_blog-src/_posts/2016-04-08-data-encryption.adoc

```diff
@@ -0,0 +1,142 @@
+
+By default, the `DefaultDataEncrypter` class is the encrypter for the gateway. The encrypt and decrypt methods accept a single parameter for the string to be encrypted or decrypted, respectively. If you look at the code, you'll notice that the key is hard coded. For those of us that are looking for a more secure approach, we offer an alternative.
+
+The `AesDataEncrypter` class is a data encrypter that accepts an encrypted secret key. When it is invoked, it expects a configuration object that contains a previously defined encrypted secret key.
+
+At this point you may be wondering how secret keys are obtained. A factory class called `ConfigFactory` is responsible for creating a configuration object using `apiman.properties`. This object is later passed to `AesDataEncrypter` as a parameter, along with the string of text to be encrypted or decrypted.
+
+== Setting the Gateway to use a Custom Encrypted Secret Key
+
+*We highly recommend that users change the default gateway encryption class* to the more secure `AesDataEncrypter` class. To do so, please follow these steps:
+
+
+--
+{empty}1. Locate and open the `apiman.properties` file.
+
+{empty}2. Uncomment the following lines, located toward the top of the file (around line 10), by removing the hashtag at the beginning of each line:
+
+```properties
+apiman.encrypter.type=io.apiman.common.util.crypt.AesDataEncrypter
+apiman.encrypter.type.secretKey=
+```
+
+{empty}3. Add your encrypted secret key.
+
+{empty}4. Save and close the file.
+
+{empty}5. $$$ Profit!
+--
+
+
+== Writing Your Own Encryption Plugin
+
+One of the major benefits of working with an open source API management platform, and open source projects in general, is the ability to read and modify the source code to make it work for your business needs. In apiman, we make it easy for you to write your own implementation of the data encrypter by http://www.apiman.io/latest/developer-guide.html#_creating_a_plugin[creating a plugin that encrypts data].
+
+Once your basic plugin is set up, you can then begin working on the Java component of your plugin that contains the functionality.
+
+Every encryption plugin must implement the `IDataEncrypter` interface like this:
+
+```java
+public class SuperDataEncrypter implements IDataEncrypter {
+  /**
+  * Constructor.
+  */
+  public SuperDataEncrypter(Map<String, String> config) {
+    thing = config.get("thing"); //$NON-NLS-1$
+
+    if (thing == null) {
+      throw new RuntimeException("Missing configuration property: apiman-manager.config.thing"); //$NON-NLS-1$
+    }
+  }
+}
+```
+
+
+Here, we are passing a configuration object that gets created in the `ConfigFactory` we discussed above. Once the `SuperDataEncrypter` class is invoked, the object is checked for a "thing" key. If it is null, a runtime exception is thrown. Pretty, straightforward so far, right?
```

Comment:
As you can see, the class is constructed with the configuration Map we discussed above.  Once the `SuperDataEncrypter` class is built, the configuration Map is checked for a "thing" key. If it is null, a runtime exception is thrown. Pretty, straightforward so far, right?